src/components/UserSelector.tsx

```diff
@@ -9,8 +27,14 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setShowUserList(!showUserList)}
+          onBlur={() => setTimeout(() => setShowUserList(!showUserList), 200)}
```

Comment:
Using `setTimeout` with `onBlur` to toggle the dropdown might lead to unexpected behavior, especially if the user interacts with the dropdown quickly. Consider using a more reliable method to manage the dropdown state, such as handling clicks outside the component.